pkg/controller/jobs/pod/pod_controller_test.go

```diff
@@ -0,0 +1,489 @@
+						Status:             metav1.ConditionTrue,
+						LastTransitionTime: metav1.Now(),
+						Reason:             "Preempted",
+						Message:            "Preempted to accommodate a higher priority Workload",
+					}).
+					Obj(),
+			},
+			wantWorkloads: []kueue.Workload{
+				*utiltesting.MakeWorkload("job", "ns").Finalizers(kueue.ResourceInUseFinalizerName).
+					PodSets(*utiltesting.MakePodSet(kueue.DefaultPodSetName, 1).Request(corev1.ResourceCPU, "1").Obj()).
+					Queue("test-queue").
+					Condition(metav1.Condition{
+						Type:               kueue.WorkloadEvicted,
+						Status:             metav1.ConditionTrue,
+						LastTransitionTime: metav1.Now(),
+						Reason:             "Preempted",
+						Message:            "Preempted to accommodate a higher priority Workload",
+					}).
+					Obj(),
+			},
+			workloadCmpOpts: defaultWorkloadCmpOpts,
+		},
+		"pod is finalized when it's succeeded": {
+			pod: *basePodWrapper.
+				Clone().
+				SetLabel("kueue.x-k8s.io/managed", "true").
+				KueueFinalizer().
+				StatusPhase(corev1.PodSucceeded).
+				Obj(),
+			wantPod: *basePodWrapper.
+				Clone().
+				SetLabel("kueue.x-k8s.io/managed", "true").
+				StatusPhase(corev1.PodSucceeded).
+				Obj(),
+			workloads: []kueue.Workload{
+				*utiltesting.MakeWorkload("unit-test", "ns").Finalizers(kueue.ResourceInUseFinalizerName).
+					PodSets(*utiltesting.MakePodSet(kueue.DefaultPodSetName, 1).Request(corev1.ResourceCPU, "1").Obj()).
+					ReserveQuota(utiltesting.MakeAdmission("cq").AssignmentPodCount(1).Obj()).
+					Admitted(true).
+					Obj(),
+			},
+			wantWorkloads: []kueue.Workload{
+				*utiltesting.MakeWorkload("unit-test", "ns").Finalizers(kueue.ResourceInUseFinalizerName).
+					PodSets(*utiltesting.MakePodSet(kueue.DefaultPodSetName, 1).Request(corev1.ResourceCPU, "1").Obj()).
+					ReserveQuota(utiltesting.MakeAdmission("cq").AssignmentPodCount(1).Obj()).
+					Admitted(true).
+					Condition(metav1.Condition{
+						Type:    "Finished",
+						Status:  "True",
+						Reason:  "JobFinished",
+						Message: "Job finished successfully",
+					}).
+					Obj(),
+			},
+			workloadCmpOpts: append(
+				defaultWorkloadCmpOpts,
+				// This is required because SSA doesn't work properly for the fake k8s client.
+				// Reconciler appends "Finished" condition using SSA. Fake client will just
+				// replace all the older conditions.
+				// See: https://github.com/kubernetes-sigs/controller-runtime/issues/2341
+				cmpopts.IgnoreSliceElements(func(c metav1.Condition) bool {
+					return c.Type == "Admitted"
+				}),
+			),
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			ctx, _ := utiltesting.ContextWithLog(t)
+			clientBuilder := utiltesting.NewClientBuilder()
+			if err := SetupIndexes(ctx, utiltesting.AsIndexer(clientBuilder)); err != nil {
+				t.Fatalf("Could not setup indexes: %v", err)
+			}
+			kcBuilder := clientBuilder.WithObjects(append(tc.initObjects, &tc.pod)...)
```

Comment:
```suggestion
			kcBuilder := clientBuilder.WithObjects(tc.initObjects, &tc.pod)
```

I think we don't need to call `append`.

Reply:
I don't think that there's a way to mix those two objects. One could either pass all the parameters for a variadic function one by one or pass a single slice with all the variadic parameters. There's no way to mix both as far as I understand.